cmd/skopeo/copy.go

```diff
@@ -99,24 +103,122 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
+		// we can get the desired effect by using CopySpecificImages.
+		return copy.CopySpecificImages, nil, nil, nil
+	case strings.HasPrefix(multiArch, "sparse:"):
+		sparseArg := strings.TrimPrefix(multiArch, "sparse:")
+		platformList, instanceList, err := parseMultiArchSparse(globalOptions, sparseArg)
+		if err != nil {
+			return copy.CopySpecificImages, nil, nil, err
+		}
+		return copy.CopySpecificImages, instanceList, platformList, nil
 	default:
-		return copy.CopySystemImage, fmt.Errorf("unknown multi-arch option %q. Choose one of the supported options: 'system', 'all', or 'index-only'", multiArch)
+		return copy.CopySystemImage, nil, nil, fmt.Errorf("unknown multi-arch option %q. Choose one of the supported options: 'system', 'sparse:...', 'all', or 'index-only'", multiArch)
+	}
+}
+
+func parseMultiArchSparse(globalOptions *globalOptions, sparseArg string) ([]imgspecv1.Platform, []digest.Digest, error) {
+	var platformList []imgspecv1.Platform
+	var instanceList []digest.Digest
+	remainder := "," + sparseArg
+	for remainder != "" {
+		parseArg := func(argStart string, argEnd string, fn func(argVal string) (bool, error)) (bool, error) {
+			if newRemainder, isArg := strings.CutPrefix(remainder, ","+argStart); isArg {
+				if !isArg {
+					return false, nil
+				}
+				var argSpec string
+				if argEnd != "" {
+					var foundArgEnd bool
+					argSpec, newRemainder, foundArgEnd = strings.Cut(newRemainder, argEnd)
+					if !foundArgEnd {
+						return false, fmt.Errorf("--multi-arch=sparse:%s: end of argument marker %s not found", argStart, argEnd)
+					}
+				}
+				handled, err := fn(argSpec)
+				if err != nil {
+					return false, err
+				}
+				if handled {
+					remainder = newRemainder
+					return true, nil
+				}
+				return false, nil
+			}
+			return false, nil
+		}
+		if isSystem, err := parseArg("system", "", func(string) (bool, error) {
+			systemPlatform := imgspecv1.Platform{
+				OS:           globalOptions.overrideOS,
+				Architecture: globalOptions.overrideArch,
+				Variant:      globalOptions.overrideVariant,
+			}
+			platformList = append(platformList, systemPlatform)
+			return true, nil
+		}); err != nil {
+			return nil, nil, err
+		} else if isSystem {
+			continue
+		}
+		if isDigest, err := parseArg("digest=[", "]", func(digestSpecList string) (bool, error) {
+			for _, digestSpec := range strings.Split(digestSpecList, ",") {
+				instanceDigest, err := digest.Parse(digestSpec)
+				if err != nil {
+					return false, fmt.Errorf("while parsing instance digest %q: %w", digestSpec, err)
+				}
+				instanceList = append(instanceList, instanceDigest)
+			}
+			return true, nil
+		}); err != nil {
+			return nil, nil, err
+		} else if isDigest {
+			continue
+		}
+		if isArch, err := parseArg("arch=[", "]", func(archSpecList string) (bool, error) {
+			wantedOS := runtime.GOOS
+			if globalOptions.overrideOS != "" {
+				wantedOS = globalOptions.overrideOS
+			}
```

Comment:
If Mac users must use `--override-os=linux --multi-arch=sparse:arch=[amd64,arm64]`, that’s not really any simpler than `--multi-arch=sparse:platform=[linux/amd64,linux/arm64]`.

I think if for an `arch`, separate and distinct from `platform`, to be worth adding (extra complexity in implementation and for users to read about), it should be an arch-only filter, not restricting the OS at all.

I also don’t think we need to build that arch-only filter at all for the first version… see the top-level comment.